### PR TITLE
Improve Winget workflow

### PR DESCRIPTION
**Pull request by sitiom**
Use the [Winget Releaser](https://bittu.eu.org/docs/wr-intro) action instead. This significantly simplifies the workflow. You also just only need the `public_repo` scope on your PAT.
image

If you would like to read about this action further, the documentation is [here](https://bittu.eu.org/docs/wr-intro), and the source code is [here](https://github.com/vedantmgoyal2009/winget-releaser).

If you want to see an example of a PR created using this action, see [microsoft/winget-pkgs/pulls (Pull request has been automatically created using 🛫 WinGet Releaser)](https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+sort%3Aupdated-desc+Pull+request+has+been+automatically+created+using+%F0%9F%9B%AB+WinGet+Releaser).

### CODE REVIEW

**Comment by gnpaone** [comment]
Um, there's no necessity to move from existing solution to new one as the existing one is already pretty straightforward. You mean "simplified" in the sense just reducing the words, but it increases the loop - instead of existing mser is released, GitHub action runs and package updates, now it's mser released, GitHub action runs, the package is ran by the bot and submits to winget, the same bot works in main repo doing the job - unnecessary increase of loop + first time contributor stuff for that new repo (The workflow stuff idea initiated by me [here] (https://github.com/microsoft/winget-pkgs/discussions/56758) few months ago.). In my opinion one simple working automation is good and existing stuff is already working hassle-free.

**Comment by sitiom** [comment]
Can you clarify what you are trying to say? The new action does the same exact thing (find the new URLs, update the manifest correspondingly, and PR to Winget) as before, so there really no added "complexity" in the end. You can check the [other repos](https://github.com/vedantmgoyal2009/vedantmgoyal2009/network/dependents?package_id=UGFja2FnZS0zMDc5MjM2ODE2) using it so you could see it for yourself.

**Comment by gnpaone** [comment]
Same thing, there's no complexity in existing solution too for making it simplified 😉. For more clarity, u can take some package that's using this method as an example, let's say "X" package, go to X package developer original repository, there GitHub action is ran with vedantmgoyal2009/vedantmgoyal2009, then it submits to vedantmgoyal2009 repo where the bot (it's using firebase over there)(u can check actions page in that repository) to verify the package and submits to winget-pkg repository where checks are ran again and moderator approves (the package to main repo is submitted by wingetbot that they created). Now mser submission is done via the GitHub action (first step as previous) then checks are ran in main winget-pkgs repo and moderator approves, direct step. U can definitely use above clues to link everything and their code is over there u can verify that.

**Comment by sitiom** [comment]
>  GitHub action is ran with vedantmgoyal2009/vedantmgoyal2009/winget-pkgs-automation/releaser-action@v1.0.0, then it submits to vedantmgoyal2009 repo

Oh, no it doesn't. You directly PR to https://github.com/microsoft/winget-pkgs like before. Example: https://github.com/microsoft/winget-pkgs/pull/60863

**Comment by gnpaone** [comment]
> 
> Example: [microsoft/winget-pkgs#60863](https://github.com/microsoft/winget-pkgs/pull/60863)

No, u still didn't get it, why do we need Winget releaser to update stuff when @rcmaehl can do it. It's in the same example of what u shown, the package from developer goes using "github-actions" to "winget releaser" then "winget releaser" to "winget-pkgs" repo as a PR instead of the package from developer submits using "github-actions"  to "winget-pkgs" repo as a PR, what's the point of extra step, not needed, and what's the "necessity" of changing from current working stuff to alternate stuff when it's doing the same thing

**Comment by sitiom** [comment]
> "github-actions" to "winget releaser" then "winget releaser" to "winget-pkgs" repo as a PR instead of the package from developer submits using "github-actions" to "winget-pkgs" repo as a PR

Winget Releaser **is** the GitHub Action, what extra step are you talking about?

> Why do we need Winget releaser to update stuff when @rcmaehl can do it.

It works **exactly** like the old action: @rcmaehl's PAT is used to PR to winget-pkgs. 

> "necessity" of changing from current working stuff to alternate stuff when it's doing the same thing - update the package, is the current way obsolete?

Like I said, it simplifies the workflow. The extra steps of [cloning the repo and getting the latest tag](https://github.com/rcmaehl/MSEdgeRedirect/blob/a7fca30b01c4da5d0bf96d23dfbc6e8526d8febb/.github/workflows/winget-submission.yml#L17-L28) is no longer needed, as Winget Releaser already does that.

**Comment by sitiom** [comment]
Maybe @vedantmgoyal2009 can help clear things up; I'm sure he'd be happy to help.

**Comment by vedantmgoyal2009** [comment]
@sitiom I will suggest to update the `uses:` to `vedantmgoyal2009/winget-releaser@latest`. I've created a separate repository for the action.

I think @gnpaone hasn't been able to understand the things clearly. I use firebase for hosting the documentation website and it isn't used by action. You can see the new separate repo to get what the action actually uses.

**Comment by sitiom** [comment]
> @sitiom I will suggest to update the `uses:` to `vedantmgoyal2009/winget-releaser@latest`. I've created a separate repository for the action.

Already did ahead of time 🙂

**Comment by vedantmgoyal2009** [comment]
Azure checks will be ran everytime whether you, me, or anyone submits a pull request at winget-pkgs repo. They are ran by Microsoft to ensure the Installer URLs in the manifests are safe and it doesn't install any viruses/malware on user's computer.

**Comment by vedantmgoyal2009** [comment]
I also manage a automation to update packages which aren't on github by getting the version and Installer URL from the software's API. Sometimes a maintainer doesn't want to manage the winget releases by themselves but they want their software to be available on winget so they request to put their repository in my automation. Here, I use a github bot ID to make "commits" so that it can be distinguished if there's some issue, etc. You may still notice that "pull requests" are created using my github id since Microsoft doesn't want to add another bot when their own @msftbot and @wingetbot are fighting over different tasks.

Hence the bot is also not used by the action anyways.

**Comment by vedantmgoyal2009** [comment]
Again, the separate repository structure may help you understand things better. I know since I was keeping everything in same repository (monorepo, kind of), you have get confused between the things: automation, it's bot and the ACTION.

**Comment by gnpaone** [comment]
> 
> I think @gnpaone hasn't been able to understand the things clearly. I use firebase for hosting the documentation website and it isn't used by action. You can see the new separate repo to get what the action actually uses.

Yah, I've seen that, buy why is it compulsory to change to this? It's kinda forcing to use this. Let people use the method they already been using na. That's why I asked in another discussion that is the current method obsolete. It's better for new winget users adopting winget releaser as they just started exploring, probably.

**Comment by vedantmgoyal2009** [comment]
Now that's @rcmaehl's decision since he's maintaining the project. I can't say anything about that. If he's fine with the current setup, he can stick with it. My action basically combines the work of getting the release tag, removing "v" from it and then the process of updating values in the manifests into one (3 -> 1 😉).

**Comment by gnpaone** [comment]
> I also manage a automation to update packages which aren't on github by getting the version and Installer URL from the software's API. Sometimes a maintainer doesn't want to manage the winget releases by themselves but they want their software to be available on winget so they request to put their repository in my automation. Here, I use a github bot ID to make "commits" so that it can be distinguished if there's some issue, etc. You may still notice that "pull requests" are created using my github id since Microsoft doesn't want to add another bot when their own @msftbot and @wingetbot are fighting over different tasks.
> 
> Hence the bot is also not used by the action anyways.

That's what, we want to manage it ourselves including package stuff like updating the package deleting the package etc etc to checking the logs of what's happening and knowing it to troubleshooting stuff here etc. This is the same reason we took chocolatey package managing from a random maintainer and using GitHub action. Chocolatey has similar stuff where a seperate repo can be used as chocolatey package repo and various packages can be pushed via that using id as you did. That's what we avoided.

**Comment by rcmaehl** [comment]
Geez. Give me a bit to read through all this

**Comment by vedantmgoyal2009** [comment]
Take your time.

File: .github/workflows/winget-submission.yml [hunk 1]
```diff
@@ -1,38 +1,15 @@
-name: Submit MSEdgeRedirect to winget
+name: Publish to WinGet
 
 on:
   release:
     types: [published]
```
Comment (vedantmgoyal2009) [inline .github/workflows/winget-submission.yml:5]:
```suggestion
    types: [released]
```

File: .github/workflows/winget-submission.yml [hunk 2]
```diff
@@ -1,38 +1,15 @@
...
   workflow_dispatch:
```
Comment (vedantmgoyal2009) [inline .github/workflows/winget-submission.yml:7]:
```suggestion

```

**Review by vedantmgoyal2009 — CHANGES_REQUESTED** [review]

File: .github/workflows/winget-submission.yml [hunk 3]
```diff
@@ -1,38 +1,15 @@
...
```
Comment (vedantmgoyal2009) [inline .github/workflows/winget-submission.yml:6]:
```suggestion
on:
  release:
    types: [released]
```

**Review by vedantmgoyal2009 — CHANGES_REQUESTED** [review]

File: .github/workflows/winget-submission.yml (hunk 3 above)
Comment (vedantmgoyal2009) [reply .github/workflows/winget-submission.yml]: Thanks @sitiom 👍

Comment (sitiom) [reply .github/workflows/winget-submission.yml]: I suggest reverting the change and letting @rcmaehl decide for himself.

**Comment by vedantmgoyal2009** [comment]
@rcmaehl any plans to merge?

**Comment by micwoj92** [comment]
no 
image

**Comment by sitiom** [comment]
> @rcmaehl any plans to merge?

Bump

**Comment by micwoj92** [comment]
@rcmaehl is slow reader

**Comment by vedantmgoyal2009** [comment]
@rcmaehl I guess you should try the improved workflow, and if you have any concerns, you can mention me or sitiom ;)

**Comment by Macleykun** [comment]
Sorry, somehow i wasn't notified about being mentioned here!
I suggest to switch over :) if it simplifies with the same ease then i believe it's worth doing so.

**Comment by vedantmgoyal2009** [comment]
> Tagging my existing winget CI contributors for their feedback. @gnpaone and @Macleykun

@rcmaehl would you like to consider this now?  One of the existing winget CI contributors has _approved_ the PR.

**Comment by rcmaehl** [comment]
> > Tagging my existing winget CI contributors for their feedback. @gnpaone and @Macleykun
> 
> @rcmaehl would you like to consider this now? One of the existing winget CI contributors has _approved_ the PR.

More than likely, working on trying to get #160 resolved at the moment.

Please let me know what I need to do to distribute only the x64 build with this new CI.

**Comment by vedantmgoyal2009** [comment]
@rcmaehl Well, there's nothing that needs to be modified distribute only x64 build.

@sitiom I would suggest incorporating the WinGet publishing step into the main CI. You can see https://github.com/rcmaehl/MSEdgeRedirect/compare/main...vedantmgoyal2009:MSEdgeRedirect:patch-1.

**Comment by sitiom** [comment]
> @sitiom I would suggest incorporating the WinGet publishing step into the main CI. You can see [main...vedantmgoyal2009:MSEdgeRedirect:patch-1](https://github.com/rcmaehl/MSEdgeRedirect/compare/main...vedantmgoyal2009:MSEdgeRedirect:patch-1).

Is that the recommended way now? The current method works and looks cleaner as it is in a separate workflow.

**Comment by gnpaone** [comment]
Yah, it's good to try new stuff as long as it works similar to current workflow 🙂. I suggest maintaining seperate workflow as main workflow contains development versions and main workflow will be running many times as per requirement

**Comment by vedantmgoyal2009** [comment]
@rcmaehl please merge 👍🏻

**Comment by rcmaehl** [comment]
@vedantmgoyal2009 

Latest action failed with an error:

```
  D:\a\_actions\vedantmgoyal2009\winget-releaser\latest\dist\index.js:9765
        version || new RegExp(/(?<=v).*/g).exec(releaseInfo.tag_name)[0],
                                                                     ^
  
  TypeError: Cannot read properties of null (reading '0')
      at D:\a\_actions\vedantmgoyal2009\winget-releaser\latest\dist\index.js:9765:68
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by micwoj92** [comment]
@vedantmgoyal2009 how about you unfuck your profile so it's viewable in browser?

**Comment by sitiom** [comment]
Possibly the regex `(?<=v).*` does not account for tags without the `v` prefix:
image

@vedantmgoyal2009

**Comment by rcmaehl** [comment]
> Possibly the regex `(?<=v).*` does not account for tags without the `v` prefix: image
> 
> @vedantmgoyal2009
> 
> At the time this was merged, the default regex was `[0-9.]+` though 🤔

It's not using version pinning in the PR

**Pull request merged**